exp/examine/latest.go

```diff
@@ -0,0 +1,94 @@
+package examine
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/chanzuckerberg/fogg/errs"
+	"github.com/spf13/afero"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+//LatestModuleVersions retrieves the latest version of the provided modules
+func LatestModuleVersions(fs afero.Fs, modules []ModuleWrapper) ([]ModuleWrapper, error) {
+	var latestModules []ModuleWrapper
+	var module ModuleWrapper
+	var resource string
+	var err error
+
+	for _, mod := range modules {
+		if strings.HasPrefix(mod.moduleSource, githubURL) { //If the resource is from github then retrieve from custom url
+			resource, err = generateURL(fs, mod)
+			if err != nil {
+				return nil, errs.WrapUserf(err, "Could not generate url for %s latest module", mod.moduleSource)
+			}
+			module.module, err = GetFromGithub(fs, resource)
+			if err != nil {
+				return nil, err
+			}
+
+			latestModules = append(latestModules, module)
+		} else if strings.HasPrefix(module.moduleSource, awsRegistry) && false { //DISABLED
+			//TODO:(EC) Enable
+			resource = module.moduleSource
+			res, err := http.Get(resource)
+			if err != nil {
+				return nil, err
+			}
+			defer res.Body.Close()
+
+			body, err := ioutil.ReadAll(res.Body)
+			if err != nil {
+				return nil, err
+			}
+
+			//FIXME: unmarshal body into ModuleWrapper, specifically in the RegistryModule part
+			//TODO:(EC) Get input about adding RegistryModule as a part of module
+			err = json.Unmarshal(body, &module)
+			if err != nil {
+				return nil, err
+			}
+			latestModules = append(latestModules, module)
+		}
+	}
+
+	return latestModules, nil
+}
+
+//generateURL creates github url for the given module
+func generateURL(fs afero.Fs, module ModuleWrapper) (string, error) {
+	url := ""
+	if strings.HasPrefix(module.moduleSource, "github.com/chanzuckerberg") { //Clone the repo and get the latest tag
```

Comment:
Don't clone the repo - just use the github api – https://godoc.org/github.com/google/go-github/github#RepositoriesService.GetLatestRelease

Reply:
Can't believe I missed this, thank you!